contents/blog/feature-flags-vs-configuration.md

```diff
@@ -0,0 +1,135 @@
+tags:
+ - Guides
+ - Feature flags
+ - Explainers
+---
+
+Sometimes you want to change the behavior of your app without changing the code. Both feature flags and configurations are options for doing this. 
+
+In their most basic form, they are a key you add to your code that returns a value you can control. You can use this value to change the behavior or experience of your app or product.
+
+This post goes over what configuration and feature flags do, and which one you should choose.
+
+## What is a configuration?
+
+A configuration, or config, is literally the definition above. A key when called with a service returns a value. You set both the key and value, then add the key and service to your code. You can then modify the key’s value. Configurations are often separated into a dedicated file or even handled by an external service. 
+
+An example of a configuration is a secret key for an external service like Stripe or AWS set in an environment variable. It is set in a `.env` file or by using the `export` keyword in bash and accessed with a module like `dotenv` or `os`.
+
+![env file](../images/blog/feature-flags-vs-configuration/env.png)
+
+At a technical level, configurations are often set at compile or load time and don’t change. This means the app needs to be re-complied or restarted to make a change to the configuration. Because of this, developers expect they won’t change often; for example, they aren’t used to provide multiple different values during the same session.
+
+When an app deploys, configurations (and environment variables) are often handled by the service hosting. For example, Vercel, Netlify, and Render let you easily set and access environment variables, while Firebase, Azure, and AWS all have configuration management services for large, elaborate enterprise products.
+
+Developers assume configurations will be permanent. Values might change, but they plan for that key and its associated value to always be important. This means versioning is critical for some configurations. Versioning helps debug potential issues related to code controlled by configs.
+
+> **Note:** Configurations can be environment variables, but they aren’t exclusively. They can be key-value pairs stored anywhere in your stack.
+
+### What is remote configuration?
+
+A remote configuration, or remote config, is a configuration that can be set and changed remotely. This means a developer can change the value for users without needing to change any code. [Firebase](https://firebase.google.com/docs/remote-config) popularized this, but it also exists for [Unity](https://unity.com/products/remote-config) and many other services.
+
+Remote config is especially relied on for mobile apps. This is because mobile apps must go through an especially long review process, which can take 24-48 hours. If an issue pops up, this review process delays shipping a fix. If an app uses remote configs, it can mitigate (turn off) the problem feature immediately.
```

Comment:
```suggestion
Remote config is especially relied on for mobile apps. This is because mobile apps must go through an especially long review process, which can take at least 24-48 hours. If an issue pops up, this review process delays shipping a fix. If an app uses remote configs, it can mitigate (turn off) the problem feature immediately.
```